pkg/kubelet/kubelet_pods_test.go

```diff
@@ -4612,270 +4625,311 @@ func TestConvertToAPIContainerStatusesForResources(t *testing.T) {
+				Resources:          &v1.ResourceRequirements{Requests: addExtendedResource(CPU1AndMem1G), Limits: addExtendedResource(CPU1AndMem1G)},
+			},
+		},
+		"newly created Pod": {
+			State:     kubecontainer.ContainerStateCreated,
+			Resources: v1.ResourceRequirements{Limits: CPU1AndMem1GAndStorage2G, Requests: CPU1AndMem1GAndStorage2G},
+			OldStatus: v1.ContainerStatus{},
+			Expected: v1.ContainerStatus{
+				Name:               testContainerName,
+				ContainerID:        testContainerID.String(),
+				Image:              "img",
+				ImageID:            "img1234",
+				State:              v1.ContainerState{Waiting: &v1.ContainerStateWaiting{}},
+				AllocatedResources: CPU1AndMem1GAndStorage2G,
+				Resources:          &v1.ResourceRequirements{Limits: CPU1AndMem1GAndStorage2G, Requests: CPU1AndMem1GAndStorage2G},
+			},
+		},
+		"newly running Pod": {
+			Resources: v1.ResourceRequirements{Limits: CPU1AndMem1GAndStorage2G, Requests: CPU1AndMem1GAndStorage2G},
+			OldStatus: v1.ContainerStatus{
+				Name:    testContainerName,
+				Image:   "img",
+				ImageID: "img1234",
+				State:   v1.ContainerState{Waiting: &v1.ContainerStateWaiting{}},
+			},
+			Expected: v1.ContainerStatus{
+				Name:               testContainerName,
+				ContainerID:        testContainerID.String(),
+				Image:              "img",
+				ImageID:            "img1234",
+				State:              v1.ContainerState{Running: &v1.ContainerStateRunning{StartedAt: metav1.NewTime(nowTime)}},
+				AllocatedResources: CPU1AndMem1GAndStorage2G,
+				Resources:          &v1.ResourceRequirements{Limits: CPU1AndMem1GAndStorage2G, Requests: CPU1AndMem1GAndStorage2G},
+			},
+		},
+		"newly terminated Pod": {
+			State: kubecontainer.ContainerStateExited,
+			// Actual resources were different, but they should be ignored once the container is terminated.
+			Resources:          v1.ResourceRequirements{Limits: CPU1AndMem1GAndStorage2G, Requests: CPU1AndMem1GAndStorage2G},
+			AllocatedResources: &v1.ResourceRequirements{Limits: CPU2AndMem2GAndStorage2G, Requests: CPU2AndMem2GAndStorage2G},
+			OldStatus: v1.ContainerStatus{
+				Name:      testContainerName,
+				Image:     "img",
+				ImageID:   "img1234",
+				State:     v1.ContainerState{Running: &v1.ContainerStateRunning{}},
+				Resources: &v1.ResourceRequirements{Limits: CPU1AndMem1GAndStorage2G, Requests: CPU1AndMem1GAndStorage2G},
+			},
+			Expected: v1.ContainerStatus{
+				Name:        testContainerName,
+				ContainerID: testContainerID.String(),
+				Image:       "img",
+				ImageID:     "img1234",
+				State: v1.ContainerState{Terminated: &v1.ContainerStateTerminated{
+					ContainerID: testContainerID.String(),
+					StartedAt:   metav1.NewTime(nowTime),
+					FinishedAt:  metav1.NewTime(nowTime),
+				}},
+				AllocatedResources: CPU2AndMem2GAndStorage2G,
+				Resources:          &v1.ResourceRequirements{Limits: CPU2AndMem2GAndStorage2G, Requests: CPU2AndMem2GAndStorage2G},
```

Comment:
Hmm...It's not very intuitive to have resource status set for a terminated pod/container (and the "created" pod case is similar too). These fields neither reflect the desired spec or the actual status. This is basically the targeted allocated resources. Do we have to mix the two?

Reply:
For non-running containers, it's important for the scheduler to know the allocated resources, but for running containers it should prefer the actual resources. In practice, the distinction for running containers doesn't matter much, but https://github.com/kubernetes/kubernetes/issues/117765 summarizes the reasoning behind it.

Not mixing them goes back to the API questions that we went back and forth on ([summarized here](https://docs.google.com/document/d/1259qZ2cwnsFKe04eUNSC0ZV2uqz4w-_e0Fy_v6G7jA8/edit?tab=t.0#heading=h.pjrqqlyd63l4)), but would also add complexity to the resource calculation (if it's not running, use AcceptedResources, if it is running use Resources).

Conceptually, I think it makes sense to just think of the field as the resources that the container is using, or will be using if it's not running (it's a bit funky for terminal containers though).

WDYT?

Reply:
> (it's a bit funky for terminal containers though).

Yeah the fact that the pod itself is terminated and we still show resource status of the terminated container is weird. So other components will have to condition on the pod status when they calculate the resource usage anyway. 

I don't want to rehash the API discussion since know the current proposal already went through extensive reviews, but yeah it does make the meaning of the field vague and conditional :\

Feel free to resolve the comment.

Reply:
I'll take a note to pay extra attention to this in the documentation. Thanks!

Reply:
Thanks for Yuju pointing out this. I greenlit this change during the KEP review thinking about the condition, but failed to think from the other components' perspective. 

+1 on the documentation with extra attention.